nlmod/dims/layers.py

```diff
@@ -1815,8 +1815,8 @@ def remove_layer(ds, layer):
         ds = ds.copy(deep=True)
         index = layers.index(layer)
         if index == 0:
-            # lower the top to the second layer
-            ds["top"] = ds["botm"].loc[layers[1]]
+            # lower the top to the botm of the first layer
+            ds["top"] = ds["botm"].loc[layers[0]]
```

Comment:
I think the docstring needs to be changed. It says now: "Removes a layer from a Dataset, **without changing elevations of other layers**". The last part is (and was already) incorrect.

Reply:
I think the docstring is (still) correct. The top and botm of other layers are not changed. Only when the top layer is removed, the top of the new top layer (that used to be the second layer) is set to the variable `top`. The top of the second layer is not changed, only the way it is represented in ds changes.